pkg/util/fs_util.go

```diff
@@ -163,20 +170,50 @@ func GetFSFromLayers(root string, layers []v1.Layer, opts ...FSOpt) ([]string, e
 		return nil, errors.New("must supply an extract function")
 	}
 
+	var totalSize int64
+	var layerSizes []int64
+	for _, l := range layers {
+		layerSize, err := l.Size()
+		if err != nil {
+			return nil, err
+		}
+		layerSizes = append(layerSizes, layerSize)
+		totalSize += layerSize
+	}
+
+	if cfg.printExtractionProgress {
+		logrus.Infof("Extracting image layers to %s (%d bytes)", root, totalSize)
+	}
+
 	extractedFiles := []string{}
+	var extractedBytes int64
 	for i, l := range layers {
 		if mediaType, err := l.MediaType(); err == nil {
 			logrus.Tracef("Extracting layer %d of media type %s", i, mediaType)
 		} else {
 			logrus.Tracef("Extracting layer %d", i)
 		}
 
+		if cfg.printExtractionProgress {
+			logrus.Infof("Extracting layer %d %d/%d (%d%%)", i, extractedBytes, totalSize, int(math.Round(float64(extractedBytes)/float64(totalSize)*100)))
```

Comment:
Showing how many layers there are will also be helpful.

```suggestion
			logrus.Infof("Extracting layer %d of %d %d/%d (%d%%)", i, len(layers), extractedBytes, totalSize, int(math.Round(float64(extractedBytes)/float64(totalSize)*100)))
```

Reply:
I agree, but I'm on the fence about diverging from what Kaniko did beforehand. 🤔